storage_test.go

```diff
@@ -518,6 +522,30 @@ func TestWriteReportForClusterOnError(t *testing.T) {
 	// prepare connection to mocked database
 	storage := main.NewFromConnection(connection, 1)
 
+	// call the tested method
+	err := storage.WriteReportForCluster(1, 2, "foo", "", time.Now(), 42)
+	if err == nil {
+		t.Errorf("error was expected while writing report for cluster: %s", err)
+	}
+
+	// connection to mocked DB needs to be closed properly
+	checkConnectionClose(t, connection)
+
+	// check if all expectations were met
+	checkAllExpectations(t, mock)
+}
+
+// TestWriteReportForClusterWrongDriver function checks the method
+// Storage.WriteReportForCluster.
+func TestWriteReportForClusterWrongDriver(t *testing.T) {
+	// prepare new mocked connection to database
+	connection, mock := mustCreateMockConnection(t)
+
+	mock.ExpectClose()
+
+	// prepare connection to mocked database
+	storage := main.NewFromConnection(connection, WrongDatabaseDriver)
+
 	// call the tested method
 	err := storage.WriteReportForCluster(1, 2, "foo", "", time.Now(), 42)
 	if err == nil {
```

Comment:
shouldn't we expect an error if the driver is `WrongDatabaseDriver`?

Reply:
https://github.com/RedHatInsights/ccx-notification-writer/blob/222613bb1b4647412edfa855026729d9044b687e/storage.go#L411

Reply:
hmm that's good question. Probably nope ATM as most other operations are DB-independent

Reply:
it's strange that `WriteReportForCluster` won't fail if the db driver is not valid

Reply:
sorry, I read `!=` instead of `==`. In any case, in the next line the message is `"error was not expected..."` and it should be `was expected`